library/std/src/sys/unix/time.rs

```diff
@@ -293,7 +293,7 @@ mod inner {
 
     impl Instant {
         pub fn now() -> Instant {
-            Instant { t: now(libc::CLOCK_MONOTONIC) }
+            Instant { t: now(libc::CLOCK_BOOTTIME) }
```

Comment:
`CLOCK_BOOTTIME` [isn't supported on Linux <= 2.6.39](https://linux.die.net/man/2/clock_gettime), but [Rust supports Linux >= 2.6.32](https://doc.rust-lang.org/nightly/rustc/platform-support.html). This should check if `-EINVAL` is returned and fall back to `CLOCK_MONOTONIC` if so.